cob_obstacle_distance/CMakeLists.txt

```diff
@@ -33,22 +33,25 @@ generate_messages(
 # Declare catkin package
 catkin_package(
    INCLUDE_DIRS include
-   LIBRARIES
-   CATKIN_DEPENDS roscpp sensor_msgs geometry_msgs visualization_msgs message_runtime
-   DEPENDS eigen
+   LIBRARIES parsers
+   CATKIN_DEPENDS roscpp shape_msgs moveit_msgs sensor_msgs geometry_msgs visualization_msgs message_runtime
+   DEPENDS assimp
 )
 
-include_directories(include ${catkin_INCLUDE_DIRS} ${EIGEN_INCLUDE_DIRS} ${FCL_INCLUDE_DIRS} ${orocos_kdl_INCLUDE_DIRS})
+include_directories(include ${catkin_INCLUDE_DIRS} ${EIGEN_INCLUDE_DIRS} ${FCL_INCLUDE_DIRS} ${orocos_kdl_INCLUDE_DIRS} ${ASSIMP_INCLUDE_DIRS})
 
 ## Build
+add_library(parsers src/parsers/stl_parser.cpp src/parsers/mesh_parser.cpp)
+target_link_libraries(parsers ${fcl_LIBRARIES} ${catkin_LIBRARIES} ${Assimp_LIBRARIES})
+
 ## Declare a cpp executable
-add_executable(cob_obstacle_distance src/cob_obstacle_distance.cpp src/marker_shapes_interface.cpp src/shapes_manager.cpp src/distance_manager.cpp src/chainfk_solvers/advanced_chainfksolverpos_recursive.cpp)
-target_link_libraries(cob_obstacle_distance ${fcl_LIBRARIES} ${catkin_LIBRARIES} ${orocos_kdl_LIBRARIES})
+add_executable(cob_obstacle_distance src/cob_obstacle_distance.cpp src/helpers/helper_functions.cpp src/marker_shapes/marker_shapes_impl.cpp src/marker_shapes/marker_shapes_interface.cpp src/shapes_manager.cpp src/distance_manager.cpp src/chainfk_solvers/advanced_chainfksolver_recursive.cpp src/frame_to_collision.cpp)
 add_dependencies(cob_obstacle_distance cob_obstacle_distance_gencpp)
+target_link_libraries(cob_obstacle_distance parsers ${fcl_LIBRARIES} ${catkin_LIBRARIES} ${orocos_kdl_LIBRARIES})
 
 ## Install
```

Comment:
Made similar to cob_cartesian_controller/CMakeLists.txt and cob_model_identifier/CMakeLists.txt